flowmachine/tests/test_cache_utils.py

```diff
@@ -0,0 +1,396 @@
+        f"UPDATE cache.cached SET cache_score_multiplier = 1 WHERE query_id='{dl_aggregate.md5}'"
+    )
+    flowmachine_connect.engine.execute(
+        f"UPDATE cache.cached SET cache_score_multiplier = 0.5 WHERE query_id='{dl.md5}'"
+    )
+    removed_query, table_size = shrink_one(flowmachine_connect)
+    assert dl.md5 == removed_query.md5
+    assert not dl.is_stored
+    assert dl_aggregate.is_stored
+
+
+def test_shrink_to_size_does_nothing_when_cache_ok(flowmachine_connect):
+    """
+    Test that shrink_below_size doesn't remove anything if cache size is within limit."""
+    dl = daily_location("2016-01-01").store().result()
+    removed_queries = shrink_below_size(
+        flowmachine_connect, get_size_of_cache(flowmachine_connect)
+    )
+    assert 0 == len(removed_queries)
+    assert dl.is_stored
+
+
+def test_shrink_to_size_removes_queries(flowmachine_connect):
+    """
+    Test that shrink_below_size removes queries when cache limit is breached."""
+    dl = daily_location("2016-01-01").store().result()
+    removed_queries = shrink_below_size(
+        flowmachine_connect, get_size_of_cache(flowmachine_connect) - 1
+    )
+    assert 1 == len(removed_queries)
+    assert not dl.is_stored
+
+
+def test_shrink_to_size_respects_dry_run(flowmachine_connect):
+    """
+    Test that shrink_below_size doesn't remove anything during a dry run."""
+    dl = daily_location("2016-01-01").store().result()
+    dl2 = daily_location("2016-01-02").store().result()
+    removed_queries = shrink_below_size(flowmachine_connect, 0, dry_run=True)
+    assert 2 == len(removed_queries)
+    assert dl.is_stored
+    assert dl2.is_stored
+
+
+def test_shrink_to_size_dry_run_reflects_wet_run(flowmachine_connect):
+    """
+    Test that shrink_below_size dry run is an accurate report."""
+    dl = daily_location("2016-01-01").store().result()
+    dl2 = daily_location("2016-01-02").store().result()
+    shrink_to = get_size_of_table(flowmachine_connect, *dl.table_name.split(".")[::-1])
+    queries_that_would_be_removed = shrink_below_size(
+        flowmachine_connect, shrink_to, dry_run=True
+    )
+    removed_queries = shrink_below_size(flowmachine_connect, shrink_to, dry_run=False)
+    assert [q.md5 for q in removed_queries] == [
+        q.md5 for q in queries_that_would_be_removed
+    ]
+
+
+def test_shrink_to_size_uses_score(flowmachine_connect):
+    """
+    Test that shrink_below_size removes cache records in ascending score order."""
+    dl = daily_location("2016-01-01").store().result()
+    dl_aggregate = dl.aggregate().store().result()
+    flowmachine_connect.engine.execute(
+        f"UPDATE cache.cached SET cache_score_multiplier = 100 WHERE query_id='{dl_aggregate.md5}'"
+    )
+    flowmachine_connect.engine.execute(
+        f"UPDATE cache.cached SET cache_score_multiplier = 0.5 WHERE query_id='{dl.md5}'"
+    )
+    table_size = get_size_of_table(flowmachine_connect, *dl.table_name.split(".")[::-1])
```

Comment:
This pattern of passing the split table_name/schema as `*args` occurs a few times but is a little cryptic to read at first. I was wondering if it's worth to optionally lett `get_size_of_table()` accept a fully qualified table name so that we don't need to do this splitting every time?

So the following would then be both allowed:
```
get_size_of_table(connection, table_name="cells", table_schema="infrastructure")
get_size_of_table(connection, table_name="infrastructure.cells")
```
(In the second case the `table_schema` must be None, of course.)

Reply:
Yes, I like that idea!

Btw, happy to leave this to a separate PR in case it is too much hassle to include it in this one.